barretenberg/cpp/src/barretenberg/stdlib/primitives/group/cycle_group.cpp

```diff
@@ -944,85 +945,78 @@ typename cycle_group<Builder>::batch_mul_internal_output cycle_group<Builder>::_
  */
 template <typename Builder>
 typename cycle_group<Builder>::batch_mul_internal_output cycle_group<Builder>::_fixed_base_batch_mul_internal(
-    const std::span<cycle_scalar> scalars,
-    const std::span<AffineElement> base_points,
-    const std::span<AffineElement const> /*unused*/)
-    requires IsUltraArithmetic<Builder>
+    const std::span<cycle_scalar> scalars, const std::span<AffineElement> base_points)
 {
-    BB_ASSERT_EQ(scalars.size(), base_points.size());
+    BB_ASSERT_EQ(scalars.size(), base_points.size(), "Points/scalars size mismatch in fixed-base batch mul");
 
-    const size_t num_points = base_points.size();
     using MultiTableId = plookup::MultiTableId;
     using ColumnIdx = plookup::ColumnIdx;
+    using plookup::fixed_base::table;
 
     std::vector<MultiTableId> plookup_table_ids;
     std::vector<AffineElement> plookup_base_points;
     std::vector<field_t> plookup_scalars;
 
     OriginTag tag{};
-    for (size_t i = 0; i < num_points; ++i) {
-        // Merge all tags of scalars
-        tag = OriginTag(tag, scalars[i].get_origin_tag());
-        std::optional<std::array<MultiTableId, 2>> table_id =
-            plookup::fixed_base::table::get_lookup_table_ids_for_point(base_points[i]);
-        ASSERT(table_id.has_value());
```

Comment:
Clear case of incorrect `optional` usage. If the code errors when the parameter doesn't exist, that means its very much not optional!